File: prob_10.java
import java.util.*;

class prob_10 {
	public static void main(String args[]) {
		long start = System.currentTimeMillis();
		int numOfPrimes = 1;
		int sumOfPrimes = 2;
		LinkedList<Integer> primes = new LinkedList<Integer>();

		primes.add(2);

		for (int i = 3; i < 2000000; i += 2) {
			ListIterator<Integer> primesIterator = primes.listIterator();

			while (primesIterator.hasNext()) {
				if (i % primesIterator.next() == 0)
					break;
				else if (!primesIterator.hasNext()) {
					sumOfPrimes += i;
					numOfPrimes++;
					primesIterator.add(i);
				}
			}
		}

		long stop = System.currentTimeMillis();

		System.out.println("Number of primes: "+numOfPrimes+" / Sum of primes: "+sumOfPrimes+" / Time processed: "+((stop-start)/1000));
	}
}

File: prob_10.py
import time

sumOfPrimes = 2
primes = [2]

start = time.clock()

for i in range(3, 2000000, 2):
	for j in xrange(0, len(primes)):
		if i % primes[j] == 0:
			break
		elif j == len(primes) - 1:
			sumOfPrimes += i
			primes.append(i)
	
stop = time.clock()

print "The sum is %i: processed in %f seconds" % (sumOfPrimes, stop-start)